### PR TITLE
feat (packages/codemod): Add codemod to remove provider facades.

### PR DESCRIPTION
**Pull request by shaper**

### CODE REVIEW

File: packages/codemod/src/test/__testfixtures__/remove-anthropic-facade.input.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// @ts-nocheck
+import { Anthropic } from '@ai-sdk/anthropic';
+
+const anthropic = new Anthropic({
+  apiKey: 'key',
+  baseURL: 'url',
+  headers: { 'custom': 'header' }
+});
+
+const model = anthropic.chat('claude-3', {
```
Comment (shaper) [inline packages/codemod/src/test/__testfixtures__/remove-anthropic-facade.input.ts:10]: Here (and for other providers) is another example of the same issue mentioned in https://github.com/vercel/ai/pull/3641#discussion_r1838830113

While we can easily (as the codemod currently does) look for method calls on the provider instance and correct them to create a model with the provider instance we track from above, this is just the tip of the iceberg in terms of potential usages needing correction.

Whatever we think re: the other PR we should likely apply to the approach we use here. Currently the codemod does the trivial easy correction.

File: packages/codemod/src/codemods/remove-google-facade.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  return removeFacade(fileInfo, api, {
+    packageName: 'google',
+    className: 'Google',
+    createFnName: 'createGoogleGenerativeAI',
+    methodNames: [
+      'chat',
+      'generativeAI',
+      'embedding',
+      'textEmbedding',
+      'textEmbeddingModel',
```
Comment (lgrammel) [inline packages/codemod/src/codemods/remove-google-facade.ts:14]: how will these be affected?

File: packages/codemod/src/codemods/remove-openai-facade.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { API, FileInfo } from 'jscodeshift';
+import { removeFacade } from './lib/remove-facade';
+
+export default function transformer(fileInfo: FileInfo, api: API) {
+  return removeFacade(fileInfo, api, {
+    packageName: 'openai',
+    className: 'OpenAI',
+    createFnName: 'createOpenAI',
+    methodNames: ['chat', 'completion'],
```
Comment (lgrammel) [inline packages/codemod/src/codemods/remove-openai-facade.ts:9]: replacement potentially problematic for 3rd party openai compatible providers

File: packages/codemod/src/test/__testfixtures__/remove-anthropic-facade.input.ts (hunk 1 above)
Comment (lgrammel) [reply packages/codemod/src/test/__testfixtures__/remove-anthropic-facade.input.ts:10]: we should be careful about the method name changes, some are needed. the provider constructor might be enough

**Review by lgrammel — APPROVED** [review]

File: packages/codemod/src/test/__testfixtures__/remove-anthropic-facade.input.ts (hunk 1 above)
Comment (shaper) [reply packages/codemod/src/test/__testfixtures__/remove-anthropic-facade.input.ts]: Updated to only edit the ctor.

**Pull request merged**